module/localisation/RobotParticleLocalisation/src/RobotParticleLocalisation.cpp

```diff
@@ -131,66 +233,108 @@ namespace module::localisation {
-            leftSide.heading      = start_state.z();
-            leftSide.heading_var  = 0.005;
-
-            reset->hypotheses.push_back(leftSide);
-            ResetRobotHypotheses::Self rightSide;
-            // Start on goal line
-            rightSide.position     = Eigen::Vector2d(start_state.x(), -start_state.y());
-            rightSide.position_cov = Eigen::Vector2d::Constant(0.5).asDiagonal();
-            rightSide.heading      = -start_state.z();
-            rightSide.heading_var  = 0.005;
-
-            reset->hypotheses.push_back(rightSide);
-            emit<Scope::DELAY>(reset, std::chrono::seconds(1));
+            filter.model.processNoiseDiagonal = cfg["process_noise_diagonal"].as<Expression>();
+            filter.model.n_rogues             = cfg["n_rogues"].as<int>();
+            filter.model.resetRange           = cfg["reset_range"].as<Expression>();
+            filter.model.n_particles          = cfg["n_particles"].as<int>();
+
+            config.debug = cfg["debug"].as<bool>();
+
+            config.start_variance = cfg["start_variance"].as<Expression>();
+        });
+
+        on<Startup, With<FieldDescription>>().then([this](const FieldDescription& fd) {
+            // Left side penalty mark
+            config.start_state.emplace_back((-fd.dimensions.field_length / 2.0) + fd.dimensions.penalty_mark_distance,
+                                            (-fd.dimensions.field_width / 2.0),
+                                            -M_PI_2);
+
+            // Right side penalty mark
+            config.start_state.emplace_back((-fd.dimensions.field_length / 2.0) + fd.dimensions.penalty_mark_distance,
+                                            (fd.dimensions.field_width / 2.0),
+                                            M_PI_2);
+
+            // // Left side goal line
+            // config.start_state.emplace_back(
+            //     (-fd.dimensions.field_length / 2.0),
+            //     (-fd.dimensions.field_width / 2.0) + ((fd.dimensions.field_width - fd.dimensions.goal_width) / 4.0),
+            //     0.0);
+
+            // // Right side goal line
+            // config.start_state.emplace_back(
+            //     (-fd.dimensions.field_length / 2.0),
+            //     (fd.dimensions.field_width / 2.0) - ((fd.dimensions.field_width - fd.dimensions.goal_width) / 4.0),
+            //     0.0);
+
+            // Penalty shootout position
+            // config.start_state.emplace_back((-fd.dimensions.field_length / 2.0) +
+            // fd.dimensions.penalty_mark_distance,
+            //                                 0.0,
+            //                                 -M_PI);
+
+            filter.set_state(config.start_state,
+                             std::vector<Eigen::Vector3d>(config.start_state.size(), config.start_variance));
         });
     }
 
+
+    // True goal posiiton
     Eigen::Vector3d RobotParticleLocalisation::getFieldPosition(const VisionGoal& goal,
```

Comment:
note that this change breaks formatting :p 
```suggestion
    [[nodiscard]] Eigen::Vector3d RobotParticleLocalisation::getFieldPosition(const VisionGoal& goal,
```